### ISSUE TITLE
[Switch] The `dragEnd()` function needs to make sure the switch is in an allowed position

### ISSUE THREAD

**Issue by tomayac** [argyleink/gui-challenges#54]
The `dragEnd()` function, triggered by `pointerup`, needs to make sure the switch is in an allowed position. Currently it is possible to drag the switch, then without releasing the finger drag outside of the switch so that the switch is in an "in-between" visual state, and then release the finger. https://github.com/argyleink/gui-challenges/blob/fd7753b97a630717ecbc4e3201641fc8ffa1ba12/switch/index.js#L37

![image](https://user-images.githubusercontent.com/145676/129221609-ff7c8135-2eed-4479-8fa3-6f1a6995dc11.png)

**Comment by argyleink** [comment]
when i release, it doesnt stick, it slides to either side. which browser are you on? 

![Kapture 2021-08-16 at 07 35 33](https://user-images.githubusercontent.com/1134620/129581069-83e0da66-045f-410a-aa3a-41a4b7b921b7.gif)

i have a window.onpointerup event, which is when the thumb position is evaluated to decide on/off. curious which scenarios you're in that arent firing that event.

**Comment by tomayac** [comment]
Try it on an actual device. The movement you make is exactly correct and repros the behavior.

**Comment by argyleink** [comment]
which device you on? works great on my devices and emulators.


https://user-images.githubusercontent.com/1134620/129590419-a30a3242-7a94-4104-b6ef-af037752ea7c.mp4

**Comment by argyleink** [comment]
looks like an iOS bug that if the user begins the drag with vertical movement, the page is scrolled and then the pointerup event (or move events) are never fired. the beginning of the gif shows the desired ux but is also beginning the swipe horizontally. the end of the gif, showing the undesired ux, is when the drag event is competing with page scroll. 

![Kapture 2021-08-16 at 09 59 12](https://user-images.githubusercontent.com/1134620/129601325-6c57b16e-56ed-481b-9da3-9c0f89d5a08e.gif)

i think there's a lot of complex stuff happening here with touch-action, page scroll and pointer events and iOS is prioritizing the page scroll over the pointer events? i'm not sure how to fix this other than preventing document scroll when the gesture has begun from inside one of these switches. maybe touch-action can help more here, cuz it's already preventing page horizontal scroll when panning the thumb (ipad was the main offender of horizontal page scroll competing with horizontal drag of the thumb). 

have ideas? did i isolate the bug in the above gif?

**Comment by tomayac** [comment]
Yeah, it looks like preventing scrolling could help improve the situation. It’s indeed pretty complex to debug what’s going on.
